File: Server_Manager/Domain/Model/Response/Student/StudentResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Domain.Model.Response.Student
{
    public class StudentResponse
    {
        //[JsonIgnore]
        public long Id { get; set; }
        public string MaSinhVien { get; set; }
        public string FristName { get; set; }
        public string LastName { get; set; }
        public string Class { get; set; }
        public string PhoneNumber { get; set; }
        public string? Email { get; set; }
        public bool? Gender { get; set; }
        public DateOnly? Dob { get; set; }
        public string? UserName { get; set; }
    }
}

File: chamcongspit_flutter/lib/data/models/student/StudentResponse.dart
class StudentResponse {
  int? id;
  String? maSinhVien;
  String? firstName;
  String? lastName;
  String? className;
  String? phoneNumber;
  String? email;
  bool? gender;
  String? dob;
  String? userName;
  String? urlAvatar;

  StudentResponse({
    this.id,
    this.maSinhVien,
    this.firstName,
    this.lastName,
    this.className,
    this.phoneNumber,
    this.email,
    this.gender,
    this.dob,
    this.userName,
    this.urlAvatar,
  });

  StudentResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    maSinhVien = json['maSinhVien'];
    firstName = json['firstName'];
    lastName = json['lastName'];
    className = json['class'];
    phoneNumber = json['phoneNumber'];
    email = json['email'];
    gender = json['gender'];
    dob = json['dob'];
    userName = json['userName'];
    urlAvatar = json['urlAvatar'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['maSinhVien'] = maSinhVien;
    data['firstName'] = firstName;
    data['lastName'] = lastName;
    data['class'] = className;
    data['phoneNumber'] = phoneNumber;
    data['email'] = email;
    data['gender'] = gender;
    data['dob'] = dob;
    data['userName'] = userName;
    data['urlAvatar'] = urlAvatar;
    return data;
  }

  @override
  String toString() {
    return '${lastName ?? ''} ${firstName ?? ''} (${maSinhVien ?? 'N/A'})'; // Handle null values explicitly
  }
}
